## Cantons information

In [ ]:
import pandas as pd

In [3]:
cantons_pop_df = pd.read_excel('./data/data_population_canton.xlsx')
cantons_pop_df

zip_code       Name_canton  Population    0–19    20–64  65 and over  \
0        VD              Vaud      845870  184231   519663       141976   
1        VS            Valais      365844   70577   219937        75330   
2        GE            Geneva      524410  111028   326994        86388   
3        BE              Bern     1063533  203904   624823       234806   
4        FR          Fribourg      341537   74535   209478        57524   
5        SO         Solothurn      286844   55335   171450        60059   
6        NE         Neuchâtel      178291   36388   106520        35383   
7        JU              Jura       74548   15217    42651        16680   
8        BS       Basel-Stadt      200031   35405   125332        39294   
9        BL  Basel-Landschaft      298837   57976   172325        68536   
10       AG            Aargau      726894  148466   441417       137011   
11       ZH            Zurich     1605508  317627  1009978       277903   
12       GL            Glarus       42056    8072    25004         8980   
13       SH      Schaffhausen       87111   16521    51468        19122   
14       AR  Appenzell A. Rh.       56495   11654    32875        11966   
15       AI  Appenzell I. Rh.       16585    3444     9696         3445   
16       SG        St. Gallen      535114  109388   322746       102980   
17       GR        Graubünden      204888   35797   121842        47249   
18       TG           Thurgau      295220   60235   178224        56761   
19       LU           Lucerne      432744   87655   264188        80901   
20       UR               Uri       37931    7542    21992         8397   
21       SZ            Schwyz      167403   32139   102749        32515   
22       OW          Obwalden       39272    7780    23288         8204   
23       NW         Nidwalden       45016    8067    26836        10113   
24       ZG               Zug      132556   26580    81878        24098   
25       TI            Ticino      357720   62877   210178        84665   

    Sex male  Sex female  Citizenship_Swiss  Citizenship_Foreigner  \
0     415697      430173             561410                 284460   
1     182413      183431             273857                  91987   
2     253501      270909             305726                 218684   
3     522248      541285             873149                 190384   
4     170884      170653             257057                  84480   
5     143570      143274             215055                  71789   
6      87554       90737             130853                  47438   
7      36901       37647              62658                  11890   
8      97361      102670             123470                  76561   
9     146562      152275             224047                  74790   
10    365960      360934             528295                 198599   
11    800992      804516            1145802                 459706   
12     21274       20782              30989                  11067   
13     43329       43782              62442                  24669   
14     28438       28057              46455                  10040   
15      8518        8067              14519                   2066   
16    269080      266034             393155                 141959   
17    102927      101961             162697                  42191   
18    148878      146342             215329                  79891   
19    216015      216729             341873                  90871   
20     19342       18589              32289                   5642   
21     85877       81526             127559                  39844   
22     19849       19423              32835                   6437   
23     23024       21992              37232                   7784   
24     67179       65377              92423                  40133   
25    174159      183561             253794                 103926   

    Marital status_Single  Marital status_Married  Marital status_Widowed  \
0        

## Stores information

In [81]:
all_stores_df = pd.read_csv('./data/output.csv')
all_stores_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7678 entries, 0 to 7677
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        7678 non-null   int64  
 1   name      7678 non-null   object 
 2   address   7678 non-null   object 
 3   postcode  7678 non-null   object 
 4   city      7678 non-null   object 
 5   lat       7678 non-null   float64
 6   lon       7678 non-null   float64
dtypes: float64(2), int64(1), object(4)
memory usage: 420.0+ KB


### Dropping rows with unknowns


swiss_data = pd.read_csv('/Users/marisadavis/Downloads/AMTOVZ_CSV_LV95/AMTOVZ_CSV_LV95.csv',sep=";")
swiss_data.head()
swiss_postcodes_df = swiss_data[['PLZ', 'Kantonskürzel']]
swiss_postcodes_df.head()

In [8]:
all_stores_df = all_stores_df[all_stores_df['name'] != 'Unknown']
all_stores_df = all_stores_df[(all_stores_df['postcode'] != 'Unknown')]

print(all_stores_df.info())
all_stores_df.head()

<class 'pandas.core.frame.DataFrame'>
Index: 4326 entries, 2 to 7676
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        4326 non-null   int64  
 1   name      4326 non-null   object 
 2   address   4326 non-null   object 
 3   postcode  4326 non-null   object 
 4   city      4326 non-null   object 
 5   lat       4326 non-null   float64
 6   lon       4326 non-null   float64
dtypes: float64(2), int64(1), object(4)
memory usage: 270.4+ KB
None


id                name           address postcode     city  \
2     97995346                Spar    Bahnhofstrasse     8753   Mollis   
3    280130028                ALDI  Säumergutstrasse     8718  Schänis   
5    573391524  Kaufhaus Schubiger           Im Dorf     8752   Näfels   
8    862942472                Aldi          Oberdorf     8752   Näfels   
10  1774376850              Migros          Oberdorf     8752   Näfels   

          lat       lon  
2   47.094740  9.074537  
3   47.155492  9.039666  
5   47.099207  9.062775  
8   47.092433  9.058581  
10  47.093637  9.059571

## Getting canton names for postcodes to add in store data set

In [10]:
cantons_dict = {'Canton de Vaud': 'VD',
                'Genève': 'GE',
                'Canton de Fribourg': 'FR',
                'Canton de Berne': 'BE' , 
                'Canton du Valais':'VS',
                'Neuchâtel':'NE' ,
                'Jura': 'JU',
                'Kanton Solothurn': 'SO',
                'Kanton Basel-Landschaft': 'BL' ,
                'Kanton Basel-Stadt':'BS' ,
                'Kanton Aargau': 'AR', 
                'Kanton Luzern': 'LU' ,
                'Kanton Nidwalden': 'NW',
                'Kanton Obwalden':'OW' ,
                'Kanton Zug': 'ZG',
                'Kanton Uri': 'UR' ,
                'Kanton Schwyz': 'SZ',
                'Ticino': 'TI',
                'Kanton Graubünden':'GR' ,
                'Kanton St. Gallen': 'SG',
                'Kanton Zürich': 'ZH',
                'Kanton Schaffhausen': 'SH ',
                'Kanton Thurgau': 'TG',
                'Kanton Glarus': 'GL', 
                'Kanton Appenzell Ausserrhoden':'AR',
                'Kanton Appenzell Innerrhoden':'AI'}
cantons_dict_reversed = {v: k for k, v in cantons_dict.items()}


In [69]:
swiss_postcodes_df= pd.read_csv('/Users/marisadavis/Downloads/AMTOVZ_CSV_LV95/AMTOVZ_CSV_LV95.csv', sep= ';')

swiss_postcodes_df.head()

Ortschaftsname   PLZ  Zusatzziffer        Gemeindename  BFS-Nr  \
0     Aeugst am Albis  8914             0     Aeugst am Albis       1   
1         Aeugstertal  8914             2     Aeugst am Albis       1   
2           Zwillikon  8909             0  Affoltern am Albis       2   
3  Affoltern am Albis  8910             0  Affoltern am Albis       2   
4          Bonstetten  8906             0          Bonstetten       3   

  Kantonskürzel            E            N Sprache    Validity  
0            ZH  2679402.872  1235842.010      de  01.07.2008  
1            ZH  2679815.372  1237404.310      de  01.07.2008  
2            ZH  2675280.133  1238108.286      de  01.07.2008  
3            ZH  2676852.012  1236929.718      de  01.07.2008  
4            ZH  2677412.150  1241078.278      de  01.07.2008

In [27]:
swiss_postcodes_df.Kantonskürzel.unique()

array(['ZH', 'BE', 'LU', 'UR', 'SZ', 'OW', 'NW', 'GL', 'ZG', 'FR', 'SO',
       'BS', 'BL', 'SH', 'AR', 'AI', 'SG', 'GR', 'AG', 'TG', 'TI', 'VD',
       'VS', 'NE', 'GE', 'JU', nan], dtype=object)

In [29]:
swiss_postcodes_df.loc[:, 'Canton_name'] = swiss_postcodes_df['Kantonskürzel'].map(cantons_dict_reversed)

In [31]:
swiss_postcodes_df_clean=swiss_postcodes_df.dropna()

In [33]:
swiss_postcodes_df_clean = swiss_postcodes_df_clean.rename(columns={"PLZ": "postcode"})
swiss_postcodes_df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5361 entries, 0 to 5746
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Ortschaftsname  5361 non-null   object 
 1   postcode        5361 non-null   int64  
 2   Zusatzziffer    5361 non-null   int64  
 3   Gemeindename    5361 non-null   object 
 4   BFS-Nr          5361 non-null   int64  
 5   Kantonskürzel   5361 non-null   object 
 6   E               5361 non-null   float64
 7   N               5361 non-null   float64
 8   Sprache         5361 non-null   object 
 9   Validity        5361 non-null   object 
 10  Canton_name     5361 non-null   object 
dtypes: float64(2), int64(3), object(6)
memory usage: 502.6+ KB


In [35]:
all_stores_df['postcode'] = all_stores_df['postcode'].astype(int)

all_stores_df_with_cantons = all_stores_df.merge(
    swiss_postcodes_df_clean[['postcode', 'Canton_name']], 
    on='postcode', 
    how='left'
)
all_stores_df_with_cantons.head()

#Save to CSV
all_stores_df_with_cantons.to_csv('all_stores_full_clean.csv', index=False)  

## Uniformizing population dataset cantons with all stores dataset cantons


In [38]:
#cantons_pop_df['Name_canton'].unique()
cantons_dict2 = {
    'Canton de Vaud': 'Vaud',
    'Genève': 'Geneva',
    'Canton de Fribourg': 'Fribourg',
    'Canton de Berne': 'Bern',
    'Canton du Valais': 'Valais',
    'Neuchâtel': 'Neuchâtel',
    'Jura': 'Jura',
    'Kanton Solothurn': 'Solothurn',
    'Kanton Basel-Landschaft': 'Basel-Landschaft',
    'Kanton Basel-Stadt': 'Basel-Stadt',
    'Kanton Aargau': 'Aargau',
    'Kanton Luzern': 'Lucerne',
    'Kanton Nidwalden': 'Nidwalden',
    'Kanton Obwalden': 'Obwalden',
    'Kanton Zug': 'Zug',
    'Kanton Uri': 'Uri',
    'Kanton Schwyz': 'Schwyz',
    'Ticino': 'Ticino',
    'Kanton Graubünden': 'Graubünden',
    'Kanton St. Gallen': 'St. Gallen',
    'Kanton Zürich': 'Zurich',
    'Kanton Schaffhausen': 'Schaffhausen',
    'Kanton Thurgau': 'Thurgau',
    'Kanton Glarus': 'Glarus',
    'Kanton Appenzell Ausserrhoden': 'Appenzell A. Rh.',
    'Kanton Appenzell Innerrhoden': 'Appenzell I. Rh.'
}
cantons_dict2_reversed = {v: k for k, v in cantons_dict2.items()}

cantons_pop_df.loc[:, 'Canton_name'] = cantons_pop_df['Name_canton'].map(cantons_dict2_reversed)

cantons_pop_df = cantons_pop_df.drop(columns=['Name_canton','zip_code'])
cantons_pop_df.head(10)

Population    0–19   20–64  65 and over  Sex male  Sex female  \
0      845870  184231  519663       141976    415697      430173   
1      365844   70577  219937        75330    182413      183431   
2      524410  111028  326994        86388    253501      270909   
3     1063533  203904  624823       234806    522248      541285   
4      341537   74535  209478        57524    170884      170653   
5      286844   55335  171450        60059    143570      143274   
6      178291   36388  106520        35383     87554       90737   
7       74548   15217   42651        16680     36901       37647   
8      200031   35405  125332        39294     97361      102670   
9      298837   57976  172325        68536    146562      152275   

   Citizenship_Swiss  Citizenship_Foreigner  Marital status_Single  \
0             561410                 284460                 418325   
1             273857                  91987                 163137   
2             305726                 218684                 261339   
3             873149                 190384                 478542   
4             257057                  84480                 162486   
5             215055                  71789                 123185   
6             130853                  47438                  83895   
7              62658                  11890                  33430   
8             123470                  76561                 100761   
9             224047                  74790                 124917   

   Marital status_Married  Marital status_Widowed  Marital status_Divorced  \
0                  315182                   33595                    76804   
1                  149547                   18664                    33968   
2                  189663                   19926                    51833   
3                  432439                   54030                    96804   
4                  134688                   13835                    30021   
5                  122480                   14062                    26671   
6                   65266                    9108                    19777   
7                   29406                    4374                     7223   
8                   69830                    9334                    19440   
9                  130738                   15879                    26752   

   Marital status_Unmarried  In a registered partnership  \
0                        90                         1292   
1                        44                          310   
2                       101                         1148   
3                        71                         1148   
4                        35                          320   
5                        15                          313   
6                        12                          166   
7                        12                           60   
8                        22                          484   
9                        17                          416   

   Partnership dissolved  Urban core area   Area influenced by urban cores  \
0                    444            521241                          236125   
1                    114            201600                           87453   
2                    368            477653                           46757   
3                    442            559799                          236233   
4                    119            111017                          142877   
5                     88            172737                           73426   
6                     63            104433                           52086   
7                     25             25116                           15040   
8                    152            200031                               0   
9                     98            203068                           88782   

   Area beyond urban core influence              Canton_name  
0                             88504      

In [40]:
#Save to CSV
cantons_pop_df.to_csv('cantons_population_clean_uniform.csv', index=False)  

## Count supermarkets per cantons and proportion of Migros

In [65]:
list_of_cantons = cantons_pop_df['Canton_name'].unique()
len(list_of_cantons)


26

## Missing canton names

I noticed there are about 500 missing canton names but these are stores in other countries

In [79]:
missing_cantons = all_stores_df_with_cantons[all_stores_df_with_cantons['Canton_name'].isna()]
missing_cantons.tail(10)

id                             name              address  \
8655   701492791                             Lidl  Corso Nazioni Unite   
8656   702102818                            Sigma   Via Alfredo di Dio   
8659   766893803                             Dpiù          Via Elvezia   
8660  1078589015                        Carrefour    Via Paolo Zaccheo   
8661  1232391465                            PENNY             Via Zara   
8662  1256843041                           Tigros     Via Delle Dogane   
8663  1266057179                           Tigros      Via Ugo Foscolo   
8664  1272285080                               U2      Via Provinciale   
8665  1278749136  Supermercato Tigros di Cuveglio   Via Giuseppe Verdi   
8666  1300760341               Dpiù di Marchirolo          Via Statale   

      postcode                city        lat       lon Canton_name  
8655     28922            Verbania  45.928454  8.551589         NaN  
8656     28877           Ornavasso  45.970584  8.414436         NaN  
8659     21050             Unknown  45.836756  8.908055         NaN  
8660     28822            Cannobio  46.063288  8.698465         NaN  
8661     22100                Como  45.836139  9.064766         NaN  
8662     21016               Luino  45.997077  8.742347         NaN  
8663     22029  Uggiate con Ronago  45.823445  8.950206         NaN  
8664     21030     Rancio Valcuvia  45.914442  8.759134         NaN  
8665     21030            Cuveglio  45.902432  8.734853         NaN  
8666     21030          Marchirolo  45.941045  8.827448         NaN

## How many Migros stores in Switzerland?

In [63]:
list_of_cantons_stores = all_stores_df_with_cantons['Canton_name'].unique()
len(list_of_cantons_stores)

25

In [46]:
all_stores_df_with_cantons.head()

id                name           address  postcode     city  \
0    97995346                Spar    Bahnhofstrasse      8753   Mollis   
1   280130028                ALDI  Säumergutstrasse      8718  Schänis   
2   573391524  Kaufhaus Schubiger           Im Dorf      8752   Näfels   
3   862942472                Aldi          Oberdorf      8752   Näfels   
4  1774376850              Migros          Oberdorf      8752   Näfels   

         lat       lon        Canton_name  
0  47.094740  9.074537      Kanton Glarus  
1  47.155492  9.039666  Kanton St. Gallen  
2  47.099207  9.062775      Kanton Glarus  
3  47.092433  9.058581      Kanton Glarus  
4  47.093637  9.059571      Kanton Glarus

In [48]:
all_stores_df_with_cantons = all_stores_df_with_cantons.replace('ALDI', 'Aldi')
all_stores_df_with_cantons = all_stores_df_with_cantons.replace('COOP', 'Coop')

all_stores_df_with_cantons = all_stores_df_with_cantons.drop_duplicates(keep='last')

In [52]:
stores_sorted_counts = all_stores_df_with_cantons.groupby('name').size().sort_values(ascending=False)
stores_sorted_counts.head(20)

name
Migros              494
Coop                463
Denner              279
Volg                209
Aldi                166
Lidl                146
Landi                85
Spar                 74
VOI                  54
Alnatura             16
Migros Partner       14
M-Outlet             14
ALDI Süd             13
Carrefour Market     11
Denner Partner        9
REWE                  9
Sutterlüty            8
Denner Express        8
PENNY                 8
Tigros                8
dtype: int64

There are 494 Migros shops in Switzerland, 14 M-Outlets, and 14 Migros Partners
(Denner is also associated with Migros since 2007. There are 279 Denner shops, 9 denner partners and 8 denner express shops)
The biggest conccurence for Migros in terms of number or stores in Switzerland are Coop with 463 shops, Volg with 209, Aldi with 166, and Lidl with 146.

For this research, we will focus on the Migros flagship store in comparison with the other flagship stores.

Denner will be treated as a competitor even though Migros got 70 percent of capital actions from Denner SA in 2007, because we want to see where to expand the Migros offer.

In [77]:
cantons_sorted_counts = all_stores_df_with_cantons.groupby('Canton_name').size().sort_values(ascending=False)
print(len(cantons_sorted_counts))
cantons_sorted_counts

24


Canton_name
Kanton Zürich                    543
Canton de Berne                  268
Kanton Luzern                    193
Kanton St. Gallen                190
Canton de Vaud                   135
Kanton Thurgau                    84
Ticino                            74
Kanton Basel-Landschaft           73
Canton de Fribourg                63
Kanton Schwyz                     62
Kanton Graubünden                 62
Kanton Solothurn                  59
Kanton Basel-Stadt                57
Genève                            50
Kanton Zug                        35
Kanton Nidwalden                  34
Kanton Obwalden                   31
Kanton Appenzell Ausserrhoden     29
Kanton Uri                        23
Neuchâtel                         17
Canton du Valais                  17
Kanton Glarus                     14
Kanton Appenzell Innerrhoden      10
Jura                               9
dtype: int64

In [141]:
def migros_proportion_by_all_stores(list_of_cantons):
    all_store_count = {}
    migros_store_count = {}
    migros_proportion = {}

    for canton in list_of_cantons:
        canton_count = all_stores_df_with_cantons[all_stores_df_with_cantons['Canton_name'] == canton].shape[0]
        all_store_count[canton] = canton_count

        migros_count = all_stores_df_with_cantons[(all_stores_df_with_cantons['Canton_name'] == canton) & (all_stores_df_with_cantons['name'] == 'Migros')].shape[0]
        migros_store_count[canton] = migros_count

        if canton_count > 0:
            migros_proportion[canton] = migros_count / canton_count * 100  
        else:
            migros_proportion[canton] = 0    
            
        #print(f"In {canton}, there are {canton_count} supermarkets, out of which {migros_count} are Migros stores.\nSo, {migros_proportion[canton]:.2f}% of shops are Migros.")

    df = pd.DataFrame({'Canton': list_of_cantons,
        'Canton_count': [all_store_count[c] for c in list_of_cantons],
        'Migros_count': [migros_store_count[c] for c in list_of_cantons],
        'Proportion_Migros': [migros_proportion[c] for c in list_of_cantons]
    })
    
    #df.to_csv('migros_proportion_by_all_stores.csv', index=False)  
    
    return migros_store_count

migros_proportion_by_all_stores(list_of_cantons)


{'Canton de Vaud': 29,
 'Canton du Valais': 4,
 'Genève': 13,
 'Canton de Berne': 45,
 'Canton de Fribourg': 7,
 'Kanton Solothurn': 17,
 'Neuchâtel': 4,
 'Jura': 1,
 'Kanton Basel-Stadt': 17,
 'Kanton Basel-Landschaft': 22,
 'Kanton Aargau': 0,
 'Kanton Zürich': 113,
 'Kanton Glarus': 3,
 'Kanton Schaffhausen': 0,
 'Kanton Appenzell Ausserrhoden': 8,
 'Kanton Appenzell Innerrhoden': 1,
 'Kanton St. Gallen': 41,
 'Kanton Graubünden': 5,
 'Kanton Thurgau': 22,
 'Kanton Luzern': 41,
 'Kanton Uri': 4,
 'Kanton Schwyz': 12,
 'Kanton Obwalden': 5,
 'Kanton Nidwalden': 6,
 'Kanton Zug': 10,
 'Ticino': 9}

Unfortunately I had to drop the rows with no postal code in the data cleaning part. This lead to having no entries for shops in the cantons of Jura and of Schaffausen. There was probably an issue with how the stores were registered in the firstplace in those two cantons. So we cannot take them into account for this analysis. A way to do it would be to refer to the city names or the geo coordinates. I will add this if I have  time at the end.

- In Canton de Vaud, there are 135 supermarkets, out of which 29 are Migros stores.
So, 21.48% of shops are Migros.

- In Canton du Valais, there are 17 supermarkets, out of which 4 are Migros stores.
So, 23.53% of shops are Migros.
- In Genève, there are 50 supermarkets, out of which 13 are Migros stores.
So, 26.00% of shops are Migros.
- In Canton de Berne, there are 268 supermarkets, out of which 45 are Migros stores.
So, 16.79% of shops are Migros.
- In Canton de Fribourg, there are 63 supermarkets, out of which 7 are Migros stores.
So, 11.11% of shops are Migros.
- In Kanton Solothurn, there are 59 supermarkets, out of which 17 are Migros stores.
So, 28.81% of shops are Migros.
- In Neuchâtel, there are 17 supermarkets, out of which 4 are Migros stores.
So, 23.53% of shops are Migros.
- In Jura, there are 9 supermarkets, out of which 1 are Migros stores.
So, 11.11% of shops are Migros.
- In Kanton Basel-Stadt, there are 57 supermarkets, out of which 17 are Migros stores.
So, 29.82% of shops are Migros.
- In Kanton Basel-Landschaft, there are 73 supermarkets, out of which 22 are Migros stores.
So, 30.14% of shops are Migros.
- In Kanton Aargau, there are 0 supermarkets, out of which 0 are Migros stores.
So, 0.00% of shops are Migros.
- In Kanton Zürich, there are 543 supermarkets, out of which 113 are Migros stores.
So, 20.81% of shops are Migros.
- In Kanton Glarus, there are 14 supermarkets, out of which 3 are Migros stores.
So, 21.43% of shops are Migros.
- In Kanton Schaffhausen, there are 0 supermarkets, out of which 0 are Migros stores.
So, 0.00% of shops are Migros.
- In Kanton Appenzell Ausserrhoden, there are 29 supermarkets, out of which 8 are Migros stores.
So, 27.59% of shops are Migros.
- In Kanton Appenzell Innerrhoden, there are 10 supermarkets, out of which 1 are Migros stores.
So, 10.00% of shops are Migros.
- In Kanton St. Gallen, there are 190 supermarkets, out of which 41 are Migros stores.
So, 21.58% of shops are Migros.
- In Kanton Graubünden, there are 62 supermarkets, out of which 5 are Migros stores.
So, 8.06% of shops are Migros.
- In Kanton Thurgau, there are 84 supermarkets, out of which 22 are Migros stores.
So, 26.19% of shops are Migros.
- In Kanton Luzern, there are 193 supermarkets, out of which 41 are Migros stores.
So, 21.24% of shops are Migros.
- In Kanton Uri, there are 23 supermarkets, out of which 4 are Migros stores.
So, 17.39% of shops are Migros.
- In Kanton Schwyz, there are 62 supermarkets, out of which 12 are Migros stores.
So, 19.35% of shops are Migros.
- In Kanton Obwalden, there are 31 supermarkets, out of which 5 are Migros stores.
So, 16.13% of shops are Migros.
- In Kanton Nidwalden, there are 34 supermarkets, out of which 6 are Migros stores.
So, 17.65% of shops are Migros.
- In Kanton Zug, there are 35 supermarkets, out of which 10 are Migros stores.
So, 28.57% of shops are Migros.
- In Ticino, there are 74 supermarkets, out of which 9 are Migros stores.
So, 12.16% of shops are Migros.

## Get proportion of Migros per population

In [147]:
def migros_proportion_by_canton_population(list_of_cantons):
    migros_store_count = migros_proportion_by_all_stores(list_of_cantons)
    migros_ratio = {}

    population_by_canton_count = {
        canton: cantons_pop_df.loc[cantons_pop_df['Canton_name'] == canton, 'Population'].values[0] for canton in list_of_cantons}

    for canton in list_of_cantons:
        # Count the number of Migros stores in the given canton
        migros_count = all_stores_df_with_cantons[
            (all_stores_df_with_cantons['Canton_name'] == canton) & 
            (all_stores_df_with_cantons['name'] == 'Migros')
        ].shape[0]

        if migros_count > 0:
            ratio = population_by_canton_count[canton] / migros_count
        else:
            ratio = None
        
        migros_ratio[canton] = ratio
        #sorted_migros_proportion = dict(sorted(migros_proportion.items(), key=lambda item: item[1], reverse=True))  # find another way because None value

    df = pd.DataFrame({'Canton': list_of_cantons,
            'Population_count': [population_by_canton_count[c] for c in list_of_cantons],
            'Migros_count': [migros_store_count[c] for c in list_of_cantons],
            'Potential_customer by shop': [migros_ratio[c] for c in list_of_cantons]
        })

        #print(f"In {canton}, there are {population_by_canton_count[canton]} inhabitants and {migros_count} Migros stores.\n"
              #f"So, 1 shop for {ratio:.2f} people." if ratio else "No Migros stores in this canton.")

    #df.to_csv('migros_proportion_by_population.csv', index=False)  

    return df

migros_proportion_by_canton_population(list_of_cantons)


Canton  Population_count  Migros_count  \
0                  Canton de Vaud            845870            29   
1                Canton du Valais            365844             4   
2                          Genève            524410            13   
3                 Canton de Berne           1063533            45   
4              Canton de Fribourg            341537             7   
5                Kanton Solothurn            286844            17   
6                       Neuchâtel            178291             4   
7                            Jura             74548             1   
8              Kanton Basel-Stadt            200031            17   
9         Kanton Basel-Landschaft            298837            22   
10                  Kanton Aargau            726894             0   
11                  Kanton Zürich           1605508           113   
12                  Kanton Glarus             42056             3   
13            Kanton Schaffhausen             87111             0   
14  Kanton Appenzell Ausserrhoden             56495             8   
15   Kanton Appenzell Innerrhoden             16585             1   
16              Kanton St. Gallen            535114            41   
17              Kanton Graubünden            204888             5   
18                 Kanton Thurgau            295220            22   
19                  Kanton Luzern            432744            41   
20                     Kanton Uri             37931             4   
21                  Kanton Schwyz            167403            12   
22                Kanton Obwalden             39272             5   
23               Kanton Nidwalden             45016             6   
24                     Kanton Zug            132556            10   
25                         Ticino            357720             9   

    Potential_customer by shop  
0                 29167.931034  
1                 91461.000000  
2                 40339.230769  
3                 23634.066667  
4                 48791.000000  
5                 16873.176471  
6                 44572.750000  
7                 74548.000000  
8                 11766.529412  
9                 13583.500000  
10                         NaN  
11                14208.035398  
12                14018.666667  
13                         NaN  
14                 7061.875000  
15                16585.000000  
16                13051.560976  
17                40977.600000  
18                13419.090909  
19                10554.731707  
20                 9482.750000  
21                13950.250000  
22                 7854.400000  
23                 7502.666667  
24                13255.600000  
25                39746.666667

In [ ]:
# lists:
# all_stores_df_with_cantons
# cantons_pop_df
# list_of_cantons

In [165]:
df1=pd.read_csv('/Users/marisadavis/Desktop/Constructor_Academy/Group project - Migros/migros_proportion_by_all_stores.csv')
df2=pd.read_csv('/Users/marisadavis/Desktop/Constructor_Academy/Group project - Migros/migros_proportion_by_population.csv')

df_results = pd.DataFrame.merge(df1, df2, how='inner', left_on=['Canton','Migros_count'], right_on=['Canton','Migros_count'])
df_results = df_results.sort_values(by='Migros_count', ascending=False)
df_results

#df_results.to_csv('migros_proportion_full.csv', index=False)  


Canton  Canton_count  Migros_count  \
11                  Kanton Zürich           543           113   
3                 Canton de Berne           268            45   
16              Kanton St. Gallen           190            41   
19                  Kanton Luzern           193            41   
0                  Canton de Vaud           135            29   
9         Kanton Basel-Landschaft            73            22   
18                 Kanton Thurgau            84            22   
5                Kanton Solothurn            59            17   
8              Kanton Basel-Stadt            57            17   
2                          Genève            50            13   
21                  Kanton Schwyz            62            12   
24                     Kanton Zug            35            10   
25                         Ticino            74             9   
14  Kanton Appenzell Ausserrhoden            29             8   
4              Canton de Fribourg            63             7   
23               Kanton Nidwalden            34             6   
17              Kanton Graubünden            62             5   
22                Kanton Obwalden            31             5   
1                Canton du Valais            17             4   
20                     Kanton Uri            23             4   
6                       Neuchâtel            17             4   
12                  Kanton Glarus            14             3   
15   Kanton Appenzell Innerrhoden            10             1   
7                            Jura             9             1   
10                  Kanton Aargau             0             0   
13            Kanton Schaffhausen             0             0   

    Proportion_Migros  Population_count  Potential_customer by shop  
11          20.810313           1605508                14208.035398  
3           16.791045           1063533                23634.066667  
16          21.578947            535114                13051.560976  
19          21.243523            432744                10554.731707  
0           21.481481            845870                29167.931034  
9           30.136986            298837                13583.500000  
18          26.190476            295220                13419.090909  
5           28.813559            286844                16873.176471  
8           29.824561            200031                11766.529412  
2           26.000000            524410                40339.230769  
21          19.354839            167403                13950.250000  
24          28.571429            132556                13255.600000  
25          12.162162            357720                39746.666667  
14          27.586207             56495                 7061.875000  
4           11.111111            341537                48791.000000  
23          17.647059             45016                 7502.666667  
17           8.064516            204888                40977.600000  
22          16.129032             39272                 7854.400000  
1           23.529412            365844                91461.000000  
20          17.391304             37931                 9482.750000  
6           23.529412            178291                44572.750000  
12          21.428571             42056                14018.666667  
15          10.000000             16585                16585.000000  
7           11.111111             74548                74548.000000  
10           0.000000            726894                         NaN  
13           0.000000             87111                         NaN